### DISCUSSION TITLE
Syntax highlighting for #If

### DISCUSSION THREAD

**Discussion by zaregg** [mark-wiemer/ahkpp#66] [Ideas]
Currently the syntax highlighting for this #if statement is like this:
![image](https://user-images.githubusercontent.com/57303863/110513511-997fab80-8106-11eb-9d21-24bc05845f0c.png)
In the AutoHotkey documentation, however, it's like this:
![image](https://user-images.githubusercontent.com/57303863/110513556-a56b6d80-8106-11eb-861a-65b01f5cdd07.png)
It would be very nice to see this kinda syntax highlighting also implemented in this extension :D

**Comment by FuPeiJiang** [comment]
omg ty, it hadn't occurred to me that AutoHotkey documentation did syntax highlighting, I could totally use that code.
___
> In the AutoHotkey documentation

where ? can you link ?

can you please paste the code in TEXT instead of only IMAGE ?:
as such : (I used this cool stuff called [Vis2 OCR](https://www.autohotkey.com/boards/viewtopic.php?f=6&t=36047#p165931)) image to text
```autohotkey
#If WinActive("ahk_class Notepad”) or WinActive(MyWindowTitle)
#Space::MsgBox You pressed Win+Spacebar in Notepad or %MyWindowTitle%.
```
you never know how it might be useful:
* for google search (text is searchable, images aren't)
* for me searching in open source `AutoHotkey documentation` source: https://github.com/Lexikos/AutoHotkey_L-Docs/tree/master/docs for `MyWindowTitle` so I can actually find where it is. and how they do it.

https://github.com/Lexikos/AutoHotkey_L-Docs/search?q=MyWindowTitle
wow Autohotkey documentation really is cool, I searched `MyWindowTitle` and found this:
https://www.autohotkey.com/docs/commands/_If.htm
(I looked manually in https://www.autohotkey.com/docs/commands/_IfWinActive.htm and didn't find it)
___
I want to know how THEY did it, because it's a website so it's surely javascript, and it's open source !
I have found it. but rip understanding the [code](https://github.com/Lexikos/AutoHotkey_L-Docs/blob/1df413766b8cdfd30d126b7f3b5c85346b8e9353/docs/static/content.js#L1656) lol


using inspect element, I found that the red is #A31515, I cloned the repo and searched for #A31515
what makes strings red : https://github.com/Lexikos/AutoHotkey_L-Docs/blob/1df413766b8cdfd30d126b7f3b5c85346b8e9353/docs/static/content.css#L786
`pre > .str`
`<pre>` with child that has class `str`

select all `<pre>` : https://github.com/Lexikos/AutoHotkey_L-Docs/blob/1df413766b8cdfd30d126b7f3b5c85346b8e9353/docs/static/content.js#L1559

all code snippets are in `<pre>` as you can see right here : https://github.com/Lexikos/AutoHotkey_L-Docs/blob/1df413766b8cdfd30d126b7f3b5c85346b8e9353/docs/commands/_If.htm#L30


syntax highlighting function: addSyntaxColors : https://github.com/Lexikos/AutoHotkey_L-Docs/blob/1df413766b8cdfd30d126b7f3b5c85346b8e9353/docs/static/content.js#L1656
it assigns `<span>` and corresponding css classes for colors I guess
this function definitely needs to be pseudocoded/explained.

you can ask for clarifications or try console.log/breakpoints
or me when I'm bored

**Reply by zaregg** [reply]
I'm not really good at JS or CSS so I've got to admit that I have no idea what is going on in that code lol. I can only give u the correct page and feedback, sorry :(. So u used the wrong Doc page for reference, I linked the correct page in my other reply.

**Reply by FuPeiJiang** [reply]
I'm interested in this (ahk parsing), ima try to turn this into pseudocode
then this could probably be used in this extension

**Reply by zaregg** [reply]
Great, thank you so much for ur time!

**Reply by mark-wiemer** [reply]
Yooo this is a super great find! Looks like the highlighting function is here: [addSyntaxColors in AutoHotkey_L-Docs/content.js](https://github.com/Lexikos/AutoHotkey_L-Docs/blob/1df413766b8cdfd30d126b7f3b5c85346b8e9353/docs/static/content.js#L1656)

Code doesn't look insane, it even looks like syntax highlighting might be do-able with regex alone? Indentation/formatting is another question, not sure if there is anywhere in this repo that addresses that.

**Comment by zaregg** [comment]
> where ? can you link ?

here is the link to the docs I screenshotted: [Commands_If](https://www.autohotkey.com/docs/commands/_If.htm)

> Can you please paste the code in TEXT instead of only IMAGE ?:
as such : (I used this cool stuff called Vis2 OCR) image to text

I used Vis2 OCR and here: `#If WinActive("“ahk_class Notepad") or WinActive(MyWindowTitle)`

**Reply by FuPeiJiang** [reply]
lol, sorry, I was unclear

I meant I "used" Vis2 OCR to convert image to text, because I couldn't copy paste
you could've just copy pasted it from https://www.autohotkey.com/docs/commands/_If.htm

**Reply by zaregg** [reply]
Ah okay haha, now ive got a cool program so thanks lmao

**Reply by FuPeiJiang** [reply]
updated some links (wrong links)

**Comment by mark-wiemer** [comment]
See #69 for a tracked issue :)
